fontforge/tottf.c

```diff
@@ -3333,7 +3333,7 @@ static void setos2(struct os2 *os2,struct alltabs *at, SplineFont *sf,
 	os2->version = 3;
     if ( sf->use_typo_metrics || sf->weight_width_slope_only )
 	os2->version = 4;
-    if ( sf->os2_version!=0 )
+    if ( sf->os2_version > os2->version )
```

Comment:
Right, that is by design the 0 OS/2 version corresponds to “Automatic” in the GUI, where FontForge will chose which fits best the fields being set, but if the user selects a version explicitly it is always used.